integration/ide/bsp-server/src/BspServerTests.scala

```diff
@@ -0,0 +1,164 @@
+                s"${dep.getModule.getOrganization.split('.').mkString("/")}/${dep.getModule.getName}/$version/${dep.getModule.getName}-$version"
+              basePath(dep.getVersion) -> basePath(s"<${dep.getModule.getName}-version>")
+            }
+        }
+
+        val replaceAll = replaceAllValues(workspacePath) ++ scalaTransitiveSubstitutions ++ Seq(
+          scalaVersion -> "<scala-version>"
+        )
+
+        compareWithGsonFixture(
+          initRes,
+          fixturePath / "initialize-build-result.json",
+          replaceAll = Seq(
+            BuildInfo.millVersion -> "<MILL_VERSION>",
+            Constants.bspProtocolVersion -> "<BSP_VERSION>"
+          )
+        )
+
+        val buildTargets = buildServer.workspaceBuildTargets().get()
+        compareWithGsonFixture(
+          buildTargets,
+          fixturePath / "workspace-build-targets.json",
+          replaceAll = replaceAll
+        )
+
+        val targetIds = buildTargets.getTargets.asScala.map(_.getId).asJava
+        val metaBuildTargetId = new b.BuildTargetIdentifier(
+          (workspacePath / "mill-build").toNIO.toUri.toASCIIString.stripSuffix("/")
+        )
+        assert(targetIds.contains(metaBuildTargetId))
+        val targetIdsSubset = targetIds.asScala.filter(_ != metaBuildTargetId).asJava
+
+        val buildTargetSources = buildServer
+          .buildTargetSources(new b.SourcesParams(targetIds))
+          .get()
+        compareWithGsonFixture(
+          buildTargetSources,
+          fixturePath / "build-targets-sources.json",
+          replaceAll = replaceAll
+        )
+
+        val buildTargetDependencySources = buildServer
+          .buildTargetDependencySources(new b.DependencySourcesParams(targetIdsSubset))
+          .get()
+        compareWithGsonFixture(
+          buildTargetDependencySources,
+          fixturePath / "build-targets-dependency-sources.json",
+          replaceAll = replaceAll
+        )
+
+        val buildTargetDependencyModules = buildServer
+          .buildTargetDependencyModules(new b.DependencyModulesParams(targetIdsSubset))
+          .get()
+        compareWithGsonFixture(
+          buildTargetDependencyModules,
+          fixturePath / "build-targets-dependency-modules.json",
+          replaceAll = replaceAll
+        )
+
+        val buildTargetResources = buildServer
+          .buildTargetResources(new b.ResourcesParams(targetIds))
+          .get()
+        compareWithGsonFixture(
+          buildTargetResources,
+          fixturePath / "build-targets-resources.json",
+          replaceAll = replaceAll
+        )
+
+        val buildTargetOutputPaths = buildServer
+          .buildTargetOutputPaths(new b.OutputPathsParams(targetIds))
+          .get()
+        compareWithGsonFixture(
+          buildTargetOutputPaths,
+          fixturePath / "build-targets-output-paths.json",
+          replaceAll = replaceAll
+        )
+
+        val buildTargetJvmRunEnvironments = buildServer
```

Comment:
Let's inline all these `val`s. Right now we have to manually keep track of them and make sure we pass each one to the right `compareWithGsonFixture` call, whereas if inlined then the 1:1 relationship is clear by construction.